### PR TITLE
Release task to bump minor version

### PR DESCRIPTION
**Pull request by rachel-fenichel**
##  The basics

- [x] I branched from develop
- [x] My pull request is against develop
- [x] My code follows the [style guide](https://developers.google.com/blockly/guides/modify/web/style-guide)

### Resolves

https://github.com/google/blockly/issues/3210

### Proposed Changes

Add to `scripts` in `package.json`:
- `bump`
- `preversion`
- `postversion`

`npm run bump` will:
- check out a new branch named `bump_version`
- change the minor version according to our `major.yyyymmdd.0` convention.
- rebuild and commit all changed files + package.json.
- tag
- push the branch and tag to google/blockly

Sample from running it while pushing to rachel-fenichel/blockly:
https://github.com/google/blockly/compare/master...rachel-fenichel:bump_version?expand=1

And the resulting tag:
https://github.com/rachel-fenichel/blockly/releases/tag/v3.20191018.0

### Reason for Changes

Release automation

### Test Coverage
Ran it with the changes going to rachel-fenichel/blockly instead of google/blockly.  I also had to copy in the `postversion` script during that test, because the checkout and rebuild removed it.

### Documentation
Update release process documentation.

### Additional Information

`appengine.yaml` still needs to be updated.

### CODE REVIEW

File: gulpfile.js [hunk 1]
```diff
@@ -809,6 +810,15 @@ function getRCBranchName() {
   return 'rc_' + yyyy + '_' + mm;
 };
 
+// Helper function: get a minor version number for the day.  Format: yyyymmdd.
+function getNewMinorVersionNumber() {
```
Comment (samelhusseini) [inline gulpfile.js]: This isn't used anywhere, just checking that wasn't accidental.

File: package.json [hunk 2]
```diff
@@ -19,6 +19,10 @@
   "scripts": {
     "build": "gulp build",
     "build:debug": "gulp build-core --verbose > build-debug.log 2>&1 && tail -3 -r build-debug.log",
+    "bump": "npm version 3.$(date +'%Y%m%d').0",
+    "version": "gulp build && git add -A",
+    "preversion": "gulp preversion",
+    "postversion": "gulp postversion",
```
Comment (samelhusseini) [inline package.json]: sort alphabetically.

File: gulpfile.js (hunk 1 above)
Comment (rachel-fenichel) [reply gulpfile.js]: Good point.  Switching to `bump_version` means I stopped using this.

File: package.json (hunk 2 above)
Comment (rachel-fenichel) [reply package.json]: Done.

**Pull request merged**